src/query/storage/m3/consolidators/multi_fetch_result.go

```diff
@@ -86,25 +102,30 @@ func (r *multiResult) Close() error {
 	return nil
 }
 
-func (r *multiResult) FinalResultWithAttrs() (SeriesFetchResult,
-	[]storage.Attributes, error) {
+func (r *multiResult) FinalResultWithAttrs() (
+	SeriesFetchResult, []storagemetadata.Attributes, error,
+) {
 	result, err := r.FinalResult()
 	if err != nil {
 		return result, nil, err
 	}
 
-	attrs := make([]storage.Attributes, result.SeriesIterators.Len())
-	// TODO: add testing around here.
-	if r.dedupeMap == nil {
-		for i := range attrs {
-			attrs[i] = r.seenFirstAttrs
+	var attrs []storagemetadata.Attributes
+	seriesData := result.seriesData
+	if iters := seriesData.seriesIterators; iters != nil {
+		l := iters.Len()
+		attrs = make([]storagemetadata.Attributes, 0, l)
+		if r.dedupeMap == nil {
+			for i := 0; i < l; i++ {
+				attrs = append(attrs, r.seenFirstAttrs)
+			}
+		} else {
+			for _, res := range r.dedupeMap.list() {
+				attrs = append(attrs, res.attrs)
+			}
 		}
 	} else {
-		i := 0
-		for _, res := range r.dedupeMap {
-			attrs[i] = res.attrs
-			i++
-		}
+		attrs = []storagemetadata.Attributes{}
```

Comment:
Is the else clause necessary? 

Seems like leaving it as just nil and zero length instead of `attrs = []storagemetadata.Attributes{}` is just fine?